src/Analyzer/FullyQualifiedClassName.php

```diff
@@ -28,11 +30,19 @@ public function toString(): string
 
     public function classMatches(string $pattern): bool
     {
+        if ($this->isNotAValidPattern($pattern)) {
+            throw new InvalidPatternException("'$pattern' is not a valid class or namespace pattern. Regex are not allowed, only * and ? wildcard.");
+        }
+
         return $this->class->matches($pattern);
     }
 
     public function matches(string $pattern): bool
     {
+        if ($this->isNotAValidPattern($pattern)) {
+            throw new InvalidPatternException("'$pattern' is not a valid class or namespace pattern. Regex are not allowed, only * and ? wildcard.");
```

Comment:
i just tried something with latest main and this breaks what worked before for us.

we add rules like this:

    addRules($config, 'Client', ['App\\Client\\SomeClient\\*', ...SPECIAL_RULES_NAMESPACES]);

it now complains that this is a regex.


    analyze class set /usr/src/app/src/Client

        0/1315 [>---------------------------]   0%'App\Client\SomeClient\*' is not a valid class or namespace pattern. Regex are not allowed, only * and ? wildcard.

when i comment out this line, everything still works. is the pattern checker incorrect? or how should i separate namespaces?